integration/integration_test.go

```diff
@@ -1809,6 +1810,86 @@ type disconnectTestCase struct {
 	verifyError errorVerifier
 }
 
+type repeatingReader struct {
+	s        string
+	interval time.Duration
+}
+
+func (r repeatingReader) Read(p []byte) (int, error) {
```

Comment:
This function never returns `EOF` or any other error. How can you know if the stream is closed?